### PR TITLE
chore(forms): add resetForm and clearData to onSubmit event

### PR DESCRIPTION
**Pull request by tujoworker**

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/Examples.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const Default = () => {
     <ComponentBox>
       <Form.Handler
         defaultData={{ email: null }}
-        onSubmit={(event) => console.log('onSubmit', event)}
+        onSubmit={(data) => console.log('onSubmit', data)}
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/Examples.tsx:10]: I found one more place where a Form.Handler still does `onSubmit={(event) => console.log('onSubmit', event)}`.
https://eufemia-git-feat-on-submit-reset-eufemia.vercel.app/uilib/components/input/demos/#submit-form-with-input
https://github.com/dnbexperience/eufemia/blob/b6baeb72a8445046b87562c09c13278d25406c16/packages/dnb-design-system-portal/src/docs/uilib/components/input/Examples.tsx#L207-L209

Not sure if we want to do `onSubmit={(data) => console.log('onSubmit', data)}` there as well.

**Review by langz — APPROVED** [review]
🥳

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/Examples.tsx (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/Examples.tsx:10]: On the input, its different. There its actually an event. While here its "data".

File: packages/dnb-eufemia/src/components/input-masked/TextMask.js [hunk 2]
```diff
@@ -119,7 +119,6 @@ export default class TextMask extends React.PureComponent {
 
     const params = {
       onChange: this.onChange,
-      defaultValue: this.props.value,
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/input-masked/TextMask.js:122]: This was the origin of the "formatting issue", when calling form.reset()

**Pull request merged**